Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -121,14 +122,25 @@ callStatusCallback :: EsqDBFlow m r => CallCallbackReq -> m CallCallbackRes
 callStatusCallback req = do
   let callStatusId = req.customField.callStatusId
   _ <- QCallStatus.findById callStatusId >>= fromMaybeM CallStatusDoesNotExist
-  runTransaction $ QCallStatus.updateCallStatus callStatusId (exotelStatusToInterfaceStatus req.status) req.conversationDuration req.recordingUrl
+  runTransaction $ QCallStatus.updateCallStatus callStatusId (exotelStatusToInterfaceStatus req.status) req.conversationDuration (Just req.recordingUrl)
   return Ack
 
-directCallStatusCallback :: EsqDBFlow m r => Text -> Call.ExotelCallStatus -> Text -> Maybe Int -> m CallCallbackRes
+directCallStatusCallback :: EsqDBFlow m r => Text -> Call.ExotelCallStatus -> Maybe Text -> Maybe Int -> m CallCallbackRes
 directCallStatusCallback callSid dialCallStatus recordingUrl_ callDuration = do
   callStatus <- QCallStatus.findByCallSid callSid >>= fromMaybeM CallStatusDoesNotExist
-  recordingUrl <- parseBaseUrl recordingUrl_
-  runTransaction $ QCallStatus.updateCallStatus callStatus.id (exotelStatusToInterfaceStatus dialCallStatus) (fromMaybe 0 callDuration) recordingUrl
+  let newCallStatus = exotelStatusToInterfaceStatus dialCallStatus
+  mbrecordingUrl <- case recordingUrl_ of
+    Just recordUrl -> do
+      if recordUrl == ""
+        then throwError CallStatusDoesNotExist
```

Comment:
Same issues for BAP

Reply:
Here we expect recordingUrl with a url or null, they should not send it as empty string, handling this case